src/gamekit.cs

```diff
@@ -724,24 +726,26 @@ interface GKScore : NSSecureCoding {
 		[Async]
 		[Export ("reportLeaderboardScores:withEligibleChallenges:withCompletionHandler:")]
 		void ReportLeaderboardScores (GKLeaderboardScore[] scores, GKChallenge[] eligibleChallenges, [NullAllowed] Action<NSError> completionHandler);
-#if !MONOMAC
+
+		[NoMac]
 		[NoTV][NoWatch]
 		[Deprecated (PlatformName.iOS, 8, 0, message: "Pass 'GKPlayers' to 'ChallengeComposeController (GKPlayer [] players, string message, ...)' instead.")]
 		[iOS (7,0)]
 		[Export ("challengeComposeControllerWithPlayers:message:completionHandler:")]
 		[return: NullAllowed]
 		UIViewController ChallengeComposeController ([NullAllowed] string[] playerIDs, [NullAllowed] string message, [NullAllowed] GKChallengeComposeHandler completionHandler);
-#endif
 
 #if MONOMAC
-		[Async (ResultTypeName = "GKChallengeComposeResult")]
 		[Mac (10,10)]
+		[NoiOS][NoMacCatalyst][NoWatch][NoTV]
+		[Async (ResultTypeName = "GKChallengeComposeResult")]
 		[Export ("challengeComposeControllerWithMessage:players:completionHandler:")]
 		NSViewController ChallengeComposeController ([NullAllowed] string message, [NullAllowed] GKPlayer [] players, [NullAllowed] GKChallengeComposeHandler completionHandler);
```

Comment:
It looks like the only difference between the macOS and non-macOS definitions is the return type, and that can be reconciled with different usings at the top of the file:

```cs
#MONOMAC
using ViewController = AppKit.NSViewController;
#else
using ViewController = UIKit.UIViewController;
#endif
```

and then you remove the `#if MONOMAC` and have a single definition.